.github/workflows/main.yml

```diff
@@ -15,34 +15,122 @@ concurrency:
-                pari-version: ['pari-2.13.0', 'pari-2.15.4', 'pari-2.15.5', '2.17.2']
+                pari-version: ['2.13.0', '2.15.4', '2.15.5', '2.17.2']
+                include:
+                  - os: macos-latest
+                    python-version: '3.10'
+                    pari-version: '2.17.2' # Whatever comes with homebrew
+                  - os: macos-latest
+                    python-version: '3.13'
+                    pari-version: '2.17.2' # Whatever comes with homebrew
+                  - os: windows-latest
+                    python-version: '3.12' # Whatever comes with msys2
+                    pari-version: '2.15.4'
+                  - os: windows-latest
+                    python-version: '3.12' # Whatever comes with msys2
+                    pari-version: '2.17.2'
         env:
           LC_ALL: C
           PARI_VERSION: ${{ matrix.pari-version }}
+        defaults:
+          run:
+            shell: ${{ matrix.os == 'windows-latest' && 'msys2 {0}' || 'bash' }}
         steps:
         - name: Set up the repository
           uses: actions/checkout@v4
+        - name: Setup MSYS2
+          if: runner.os == 'Windows'
+          id: msys2
+          uses: msys2/setup-msys2@v2
+          with:
+              msystem: UCRT64
+              update: true
+              install: >-
+                base-devel
+                m4
+                bison
+                make
+                patch
+                sed
+                wget
+                mingw-w64-ucrt-x86_64-toolchain
+                mingw-w64-ucrt-x86_64-gmp
+                mingw-w64-ucrt-x86_64-python
+                mingw-w64-ucrt-x86_64-python-pip
+                mingw-w64-ucrt-x86_64-meson-python
+                mingw-w64-ucrt-x86_64-cython
+              path-type: inherit
         - name: Set up Python ${{ matrix.python-version }}
+          if: runner.os != 'Windows'
           uses: actions/setup-python@v4
           with:
             python-version: ${{ matrix.python-version }}
-        - name: Install pari
+        - name: Install PARI
+          if: matrix.os != 'macos-latest'
+          run: |         
+            bash -x .install-pari.sh
+        - name: Install PARI
+          if: matrix.os == 'macos-latest'
+          run: |
+            brew install pari
+        - name: Smoke test PARI
           run: |
-              bash -x .install-pari.sh
-        - name: Local build
+            if [ ${{ runner.os }} = macOS ]; then
+              HOMEBREW=`brew --prefix`
+              gcc -v tests/test.c -o test -I$HOMEBREW/include -L$HOMEBREW/lib -lpari -lgmp 
+              export C_INCLUDE_PATH=$HOMEBREW/include
+              export LIBRARY_PATH=$HOMEBREW/lib:$HOMEBREW/Cellar/pari/${{ matrix.pari-version }}/lib:$LIBRARY_PATH
+              gcc -v tests/test.c -o test -lpari -lgmp
+            else
+              gcc -v tests/test.c -o test -I/usr/local/include -L/usr/local/bin -lpari -lgmp
```

Comment:
I don't think you need any `-I` and `-L`  - but if you do, don't use `/usr/local`, use `$(brew --prefix)`